### ISSUE TITLE
relist date calculation is buggy

### ISSUE THREAD

**Issue by NovemLinguae** [evad37/xfdcloser#91]
Reported by Liz and @primefac at https://en.wikipedia.org/wiki/Wikipedia_talk:XFDcloser#Relist_date_calculation_bug

Steps to reproduce
(unsure)
* set computer time zone to UTC 0
* move computer to a different timezone such as UTC-8
* wait until it's almost the next day
* relist an XFD discussion that uses a log page rather than individual pages (TFD, CFD, etc.)

What happens?
* the page chosen, which has the date in it, is off by 1 day

What should happen instead?
* the page chosen should be the date exactly one week from the date of the relist

Notes
* example: https://en.wikipedia.org/w/index.php?title=Special:Contributions&offset=20240719000322&target=Primefac&limit=4
* I forget. is relisting 1 week from the moment of relist, or 1 week from the initial creation date of the entry?

**Comment by primefac** [comment]
1 week from the moment of relist based on the servers (i.e. UTC), so if one is relisting on 19 July the discussion should be put on the 19 July log page (even if their local time is still on 18 July)

**Comment by Trialpearswiki** [comment]
Given that these only seem to occur minutes after midnight I'm guessing it's something to do with the relist date being decided before midnight but used after rather than time zone issues.

**Comment by primefac** [comment]
> Given that these only seem to occur minutes after midnight I'm guessing it's something to do with the relist date being decided before midnight but used after rather than time zone issues.

Yeah, a cache issue could definitely do it, page is loaded on day X but the final button is clicked on day X+1...
